File: front-end/src/app/service-models/app/v1/client/permission-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { ModuleNameSM } from '../../enums/module-name-s-m.enum';
import { RoleTypeSM } from '../../enums/role-type-s-m.enum';

export class PermissionSM extends CoinManagementServiceModelBase<number> {
    view!: boolean;
    add!: boolean;
    edit!: boolean;
    delete!: boolean;
    isEnabledForClient!: boolean;
    moduleName!: ModuleNameSM;
    roleType!: RoleTypeSM;
    companyModulesId!: number;
    clientCompanyDetailId!: number;
    clientUserId?: number;
    licenseTypeId?: number;
    isStandAlone!: boolean;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/PermissionSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class PermissionSM : SiffrumPayrollServiceModelBase<int>
    {
        public bool View { get; set; }
        public bool Add { get; set; }
        public bool Edit { get; set; }
        public bool Delete { get; set; }
        public bool IsEnabledForClient { get; set; }
        public ModuleNameSM ModuleName { get; set; }
        public RoleTypeSM RoleType { get; set; }
        public int CompanyModulesId { get; set; }
        public int ClientCompanyDetailId { get; set; }
        public int? ClientUserId { get; set; }
        public int? LicenseTypeId { get; set; }
        public bool IsStandAlone { get; set; }
    }
}
